.github/workflows/testing.yml

```diff
@@ -10,19 +10,24 @@ jobs:
         steps:
             - name: Checkout the Git repository
               uses: actions/checkout@v2
-            - name: Install missing tools
-              run: |
-                sudo apt-get install bsdtar tree
+            - name: Setup Toltec dependencies
+              uses: ./.github/actions/setup
             - name: Build packages
               run: |
                 remote_repo="https://toltec-dev.org/testing" make repo
-            - name: Transfer packages and index
-              run: |
-                mkdir -p private
-                chmod 700 private
-                echo "${{ secrets.SSH_PRIVATE_KEY }}" > private/id_rsa
-                echo "${{ secrets.SSH_KNOWN_HOSTS }}" > private/known_hosts
-                chmod 600 private/*
-                rsync --archive --verbose --compress --delete \
-                    -e "ssh -i private/id_rsa -o UserKnownHostsFile=private/known_hosts" \
-                    build/repo/ "${{ secrets.REMOTE }}":/srv/toltec/testing
+            - name: Sync packages with the remote repository
+              uses: ./.github/actions/sync-repository
+              with:
+                local-path: build/repo/
+                ssh-key: ${{ secrets.SSH_PRIVATE_KEY }}
+                ssh-known-hosts: ${{ secrets.SSH_KNOWN_HOSTS }}
+                remote-path: ${{ secrets.REMOTE }}:/srv/toltec/testing
+            - name: Send notification to Discord
+              uses: ./.github/actions/discord-send
+              with:
+                channel: ${{ secrets.DISCORD_TESTING_CHANNEL_ID }}
+                token: ${{ secrets.DISCORD_TESTING_CHANNEL_TOKEN }}
+                title: New Toltec testing update available
+                link: https://toltec-dev.org/testing
+                color: 0xe3b341
+                message: ${{ github.event.head_commit.message }}
```

Comment:
what happens if the discord secrets are unset? (i have a repo which builds and pushes to my own URL and am wondering if the build will fail if the secrets are missing)

Reply:
It does fail the build — see <https://github.com/toltec-dev/workflow-tests/runs/1708568012>. I’ll think about a workaround.

Reply:
Okay, adding the `continue-on-error` flag makes it ignore errors from the Discord step. (See <https://github.com/toltec-dev/workflow-tests/runs/1708690940>.)